### ISSUE TITLE
[Feature Request] multiscale ssim

### ISSUE THREAD

**Issue by Ir1d** [kornia/kornia#451]
Currently SSIM is supported: https://kornia.readthedocs.io/en/latest/losses.html#kornia.losses.SSIM
Multiscale SSIM would also be nice.
example: https://github.com/jorge-pessoa/pytorch-msssim

**Comment by edgarriba** [comment]
@Ir1d great suggestion. I invite you port it to kornia. Cound the existing ssim be reused for that ?

**Comment by Ir1d** [comment]
@edgarriba I believe the existing ssim could be used.
However, when I use the linked ms-ssim as loss function, it is not very stable (sometimes nan).
I'll have to dig more into this.

**Comment by edgarriba** [comment]
@Ir1d could you file a bug to report this stability  issue ? haven't played recently with this loss, but could you try to add an epsilon value in [here](https://github.com/kornia/kornia/blob/master/kornia/losses/ssim.py#L133-L134) ?

**Comment by Ir1d** [comment]
@edgarriba , sorry about this. I meant that the linked `ms-ssim` implementation is not very stable (as I reported [here](https://github.com/jorge-pessoa/pytorch-msssim/issues/7#issuecomment-538862908)). I haven't tried implementing ms-ssim with kornia.losses.ssim yet

**Comment by Auth0rM0rgan** [comment]
Hey @Ir1d, have a look at this file [ssim.and.ms-ssim.with.1D.kernel.txt] -- haven't tested with pytorch 1.4 (https://github.com/kornia/kornia/files/4286753/ssim.and.ms-ssim.with.1D.kernel.txt)

**Comment by edgarriba** [comment]
@Ir1d @Auth0rM0rgan if any of you have time to test this function please don't hesitate to send a PR. Notice that there's an open discussion to rename the functionality #473

**Comment by v-prgmr** [comment]
> @Ir1d could you file a bug to report this stability issue ? haven't played recently with this loss, but could you try to add an epsilon value in [here](https://github.com/kornia/kornia/blob/master/kornia/losses/ssim.py#L133-L134) ?

I have stumbled on a stability issue which points me exactly to this division. I will add an epsilon for now, run some tests and get back to you

**Comment by edgarriba** [comment]
@v-prgmr thanks so much for the contribution. Notice that;s there's a pending issue to rename this functionality. Consider taking it: https://github.com/kornia/kornia/issues/473

**Comment by v-prgmr** [comment]
@edgarriba , Hi again, I would like to mention that the error that I faced with the division is in fact not anywhere from the implementation of SSIM in Kornia. However, renaming the functionality would still be a good choice as **structural dissimilarity** is the output.

**Comment by edgarriba** [comment]
@v-prgmr great to know. Would be great if you could send a small PR with this fix naming.

**Comment by v-prgmr** [comment]
@edgarriba , will submit a request asap (y)

**Comment by MohitLamba94** [comment]
> Currently SSIM is supported: https://kornia.readthedocs.io/en/latest/losses.html#kornia.losses.SSIM
> Multiscale SSIM would also be nice.
> example: https://github.com/jorge-pessoa/pytorch-msssim

The following repository I found to be best among existing MSSIM repos.
https://github.com/VainF/pytorch-msssim

**Comment by edgarriba** [comment]
@MohitLamba94 single scale version seems quite similar. Feel free to adapt multiscale version

**Comment by MohitLamba94** [comment]
Correct they simply use the same SSIM loss with same window size but downsample the image size by two and do this five times. Finally a weighted average is taken for all 5 losses !!

**Comment by edgarriba** [comment]
@v-prgmr @MohitLamba94 @Auth0rM0rgan anyone interested in contributing with this ?

**Comment by edgarriba** [comment]
If anyone interested in this please, re-open. Thanks !
